### PR TITLE
Json Schema: Stop throwing errors for user errors

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix [#2535](https://github.com/microsoft/typespec/issues/2535)

Converted most of the `throw error` to either:
- `compileAssert` if it is somethign that shouldn't happen
- `reportDiagnositc` if this is something actionable by the user

There is still 2 `throw error` as I'm not sure if those are things that could happen or not. I'll leave them until someone maybe gets a crash and we can then convert.

### CODE REVIEW

File: packages/json-schema/src/lib.ts [hunk 1]
```diff
@@ -89,6 +89,18 @@ export const libDef = {
         default: paramMessage`Invalid type '${"type"}' for a default value`,
       },
     },
+    "duplicate-id": {
+      severity: "error",
+      messages: {
+        default: paramMessage`There is multiple types with the same id "${"id"}".`,
```
Comment (markcowl) [inline packages/json-schema/src/lib.ts:95]:
```suggestion
        default: paramMessage`There are multiple types with the same id "${"id"}".`,
```

File: packages/json-schema/test/utils.ts [hunk 2]
```diff
@@ -10,21 +11,21 @@ export async function getHostForCadlFile(contents: string, decorators?: Record<s
     libraries: [JsonSchemaTestLibrary],
   });
   if (decorators) {
-    await host.addJsFile("dec.js", decorators);
+    host.addJsFile("dec.js", decorators);
```
Comment (markcowl) [inline packages/json-schema/test/utils.ts:14]: Does this not do blocking i/o?

**Review by markcowl — APPROVED** [review]
A couple of nits and a question, otherwise, lgtm

File: packages/json-schema/test/utils.ts (hunk 2 above)
Comment (timotheeguerin) [reply packages/json-schema/test/utils.ts:14]: nah, it was showing an ide "warning" that this await aws unecessary. In the test host this is just a in memory thing

**Pull request merged**